typescript/sdk/src/token/types.ts

```diff
@@ -166,3 +215,108 @@ function isCollateralRebasePairedCorrectly(
   );
   return allOthersSynthetic;
 }
+
+type CCIPContractExistsMap = ChainMap<ChainMap<boolean>>;
+
+function getCCIPConfigMaps(
+  warpRouteDeployConfig: Record<string, HypTokenRouterConfig>,
+): {
+  ccipHookMap: CCIPContractExistsMap;
+  ccipIsmMap: CCIPContractExistsMap;
+} {
+  const ccipHookMap: CCIPContractExistsMap = {};
+  const ccipIsmMap: CCIPContractExistsMap = {};
+
+  for (const [chainName, config] of Object.entries(warpRouteDeployConfig)) {
+    traverseHookConfig(chainName, config.hook, ccipHookMap);
+    traverseIsmConfig(chainName, config.interchainSecurityModule, ccipIsmMap);
+  }
+
+  return { ccipHookMap, ccipIsmMap };
+}
+
+function traverseHookConfig(
```

Comment:
This is another maybe not practical comment, but this as far as I can create a generic version of this (always forget if they are called functors or monads) and then express the extraction (i.e. the population of existsCCIPHookMap) as a generic function. Until then I would probably name this function specific to cciphook extraction